lib/internal/streams/from.js

```diff
@@ -53,18 +50,22 @@ function from(Readable, iterable, opts) {
   };
 
   readable._destroy = function(error, cb) {
-    if (needToClose) {
-      needToClose = false;
-      close().then(
-        () => process.nextTick(cb, error),
-        (e) => process.nextTick(cb, error || e),
-      );
-    } else {
-      cb(error);
-    }
+    close(error).then(
+      () => process.nextTick(cb, error), // nextTick is here in case cb throws
+      (e) => process.nextTick(cb, error || e),
```

Comment:
I think `e` should be prioritized here. Inner error should propagate over outer error.